### ISSUE TITLE
`enforce-canonical-classes` Removes spaces in custom selectors

### ISSUE THREAD

**Issue by mauriciabad** [schoero/eslint-plugin-better-tailwindcss#312]
### Description

The rule `enforce-canonical-classes` is removing too many spaces in custom selectors.

### Flavor

_Any_

### Code Input

```
[&:has(~_*_*:checked)]:opacity-100
```

### Expected Behavior

```
[&:has(~*_*:checked)]:opacity-100
```

### Actual Behavior

```
[&:has(~**:checked)]:opacity-100
```

`**` is invalid CSS.

### Reproduction URL

https://stackblitz.com/edit/vitejs-vite-w1joxqjm?file=src%2Fmain.ts

### ESLint Log

```shell
/home/projects/vitejs-vite-w1joxqjm/src/main.ts
  7:23  warning  The class: "[&:has(~_*_*:checked)]:opacity-100" can be simplified to "[&:has(~**:checked)]:opacity-100"  better-tailwindcss/enforce-canonical-classes

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.
```

### ESLint Config

_Default_

### Versions

- ESLint: 9.39.2
- Parser: 8.54.0
- Plugin: 4.0.2
- Node: v20.19.1

**Comment by schoero** [comment]
This is done by Tailwind CSS. It also happens via the VSCode Tailwind CSS Intellisense extension.

I created an issue in their repository: https://github.com/tailwindlabs/tailwindcss-intellisense/issues/1544

**Comment by schoero** [comment]
This will be fixed with the next release of Tailwind CSS: https://github.com/tailwindlabs/tailwindcss-intellisense/issues/1544#issuecomment-4323188472
